res/controllers/common-controller-scripts.js

```diff
@@ -109,561 +102,464 @@ var msecondstominutes = function(msecs) {
+                Call this each time the tap button is pressed.
+       Input:   Mixxx deck to adjust
+       Output:  -
+       -------- ------------------------------------------------------ */
+    tapButton(deck) {
+        const now = new Date() / 1000; // Current time in seconds
+        const tapDelta = now - this.tapTime;
+        this.tapTime = now;
+
+        // assign tapDelta in cases where the button has not been pressed previously
+        if (this.tap.length < 1) {
+            this.previousTapDelta = tapDelta;
         }
-        if (inputValue > currentRelative - this.maxJump
-                && inputValue < currentRelative + this.maxJump) {
-            engine.setValue(group, this.mappedFunction, outputValue);
+        // reset if longer than two seconds between taps
+        if (tapDelta > 2.0) {
+            this.tap = [];
+            return;
         }
-    } else {
-        engine.setValue(group, this.mappedFunction, outputValue);
-    }
-};
+        // reject occurrences of accidental double or missed taps
+        // a tap is considered missed when the delta of this press is 80% longer than the previous one
+        // and a tap is considered double when the delta is shorter than 40% of the previous one.
+        // these numbers are just guesses that produced good results in practice
+        if ((tapDelta > this.previousTapDelta * 1.8) || (tapDelta < this.previousTapDelta * 0.6)) {
+            return;
+        }
+        this.previousTapDelta = tapDelta;
+        this.tap.push(60 / tapDelta);
+        // Keep the last 8 samples for averaging
+        if (this.tap.length > 8) { this.tap.shift(); }
+        let sum = 0;
+        for (let i = 0; i < this.tap.length; i++) {
+            sum += this.tap[i];
+        }
+        const average = sum / this.tap.length;
 
-// Deck
+        const group = `[Channel${  deck  }]`;
 
-var Deck = function(deckNumber, group) {
-    this.deckNumber = deckNumber;
-    this.group = group;
-    this.Buttons = [];
-};
-Deck.prototype.setControlValue = Controller.prototype.setControlValue;
-Deck.prototype.addButton = Controller.prototype.addButton;
+        // "bpm" was changed in 1.10 to reflect the *adjusted* bpm, but I presume it
+        // was supposed to return the tracks bpm (which it did before the change).
+        // "file_bpm" is supposed to return the set BPM of the loaded track of the
+        // channel.
+        let fRateScale = average / engine.getValue(group, "file_bpm");
 
-// ----------------- END Object definitions ----------------------
+        // Adjust the rate:
+        fRateScale = (fRateScale - 1.) / engine.getValue(group, "rateRange");
+
+        engine.setValue(
+            group, "rate",
+            fRateScale * engine.getValue(group, "rate_dir"));
+    }
+}
+// Add class bpm to the Global JavaScript object
+// @ts-ignore Same identifier for class and instance needed for backward compatibility
+this.bpm = new bpm();
```

Comment:
wdyt of naming the class differently so it can be used in newer code? Currently mappings are forced to use the "static" API because it shadows the class.